### ISSUE TITLE
Can't import github.com/gabriel-vasile/mimetype

### ISSUE THREAD

**Issue by Trois-Six** [containous/yaegi#696]
The following program `sample.go` triggers a panic:

```go
package main

import (
  "fmt"

  "github.com/gabriel-vasile/mimetype"
)

func main() {
  file := "pdf.pdf"
  mime, err := mimetype.DetectFile(file)
  fmt.Println(mime.String(), mime.Extension(), err)
}
```

Expected result:
```console
$ yaegi ./main.go
application/pdf .pdf <nil>
```

Got:
```console
$ yaegi ./main.go
/home/perraud/src/softwares/go/src/github.com/gabriel-vasile/mimetype/mime.go:63:3: panic
/home/perraud/src/softwares/go/src/github.com/gabriel-vasile/mimetype/tree.go:8:52: panic
reflect.Set: value of type []*struct { Xmime string; Xaliases []string; Xextension string; XmatchFunc func([]uint8) bool; Xchildren []*interface {}; Xparent *interface {} } is not assignable to type []*interface {}
goroutine 14 [running]:
runtime/debug.Stack(0x1, 0xc0000d8600, 0x40)
	/usr/lib/go-1.14/src/runtime/debug/stack.go:24 +0x9d
github.com/containous/yaegi/interp.(*Interpreter).Eval.func1(0xc000a19f78)
	/home/perraud/src/softwares/go/src/github.com/containous/yaegi/interp/interp.go:330 +0xbc
panic(0xc9b420, 0xc0003c2fd0)
	/usr/lib/go-1.14/src/runtime/panic.go:969 +0x166
github.com/containous/yaegi/interp.runCfg.func1(0xc000314000, 0xc00090f000)
	/home/perraud/src/softwares/go/src/github.com/containous/yaegi/interp/run.go:114 +0x20c
panic(0xc9b420, 0xc0003c2fd0)
	/usr/lib/go-1.14/src/runtime/panic.go:969 +0x166
github.com/containous/yaegi/interp.runCfg.func1(0xc0001c4280, 0xc0000ecc00)
	/home/perraud/src/softwares/go/src/github.com/containous/yaegi/interp/run.go:114 +0x20c
panic(0xc9b420, 0xc0003c2fd0)
	/usr/lib/go-1.14/src/runtime/panic.go:969 +0x166
reflect.Value.assignTo(0xc000926fc0, 0xc00052d580, 0x197, 0xdce00f, 0xb, 0xc000926e00, 0x0, 0xc00089c2a0, 0xc0001d24e0, 0xc0001d2480)
	/usr/lib/go-1.14/src/reflect/value.go:2403 +0x426
reflect.Value.Set(0xc000926e00, 0xc0001ac280, 0x197, 0xc000926fc0, 0xc00052d580, 0x197)
	/usr/lib/go-1.14/src/reflect/value.go:1532 +0xbd
github.com/containous/yaegi/interp.doCompositeSparse.func1(0xc0001c4280, 0xc000926340)
	/home/perraud/src/softwares/go/src/github.com/containous/yaegi/interp/run.go:1943 +0x13d
github.com/containous/yaegi/interp.runCfg(0xc0000ecc00, 0xc0001c4280)
	/home/perraud/src/softwares/go/src/github.com/containous/yaegi/interp/run.go:120 +0x6d
github.com/containous/yaegi/interp.call.func7(0xc000314000, 0xc000895110)
	/home/perraud/src/softwares/go/src/github.com/containous/yaegi/interp/run.go:895 +0xc42
github.com/containous/yaegi/interp.runCfg(0xc00090f000, 0xc000314000)
	/home/perraud/src/softwares/go/src/github.com/containous/yaegi/interp/run.go:120 +0x6d
github.com/containous/yaegi/interp.(*Interpreter).run(0xc00014c1a0, 0xc00090e400, 0x0)
	/home/perraud/src/softwares/go/src/github.com/containous/yaegi/interp/run.go:98 +0x239
github.com/containous/yaegi/interp.(*Interpreter).importSrc(0xc00014c1a0, 0x0, 0x0, 0xc00003a241, 0x22, 0x0, 0x1, 0xc0000ab810, 0x540061)
	/home/perraud/src/softwares/go/src/github.com/containous/yaegi/interp/src.go:126 +0x8e9
github.com/containous/yaegi/interp.(*Interpreter).gta.func1(0xc000098f00, 0xc0000abd58)
	/home/perraud/src/softwares/go/src/github.com/containous/yaegi/interp/gta.go:178 +0x112b
github.com/containous/yaegi/interp.(*node).Walk(0xc000098f00, 0xc000a19d58, 0x0)
	/home/perraud/src/softwares/go/src/github.com/containous/yaegi/interp/interp.go:188 +0xc9
github.com/containous/yaegi/interp.(*node).Walk(0xc000098c00, 0xc000a19d58, 0x0)
	/home/perraud/src/softwares/go/src/github.com/containous/yaegi/interp/interp.go:192 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc000098a00, 0xc000a19d58, 0x0)
	/home/perraud/src/softwares/go/src/github.com/containous/yaegi/interp/interp.go:192 +0x70
github.com/containous/yaegi/interp.(*Interpreter).gta(0xc00014c1a0, 0xc000098a00, 0xc000034360, 0x4, 0x0, 0x0, 0x4, 0xc000096580, 0x0, 0x0, ...)
	/home/perraud/src/softwares/go/src/github.com/containous/yaegi/interp/gta.go:14 +0x131
github.com/containous/yaegi/interp.(*Interpreter).gtaRetry(0xc00014c1a0, 0xc000a19f20, 0x1, 0x1, 0xc000034360, 0x4, 0x0, 0x0, 0x0, 0x0)
	/home/perraud/src/softwares/go/src/github.com/containous/yaegi/interp/gta.go:239 +0x14a
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc00014c1a0, 0xc0003420c0, 0x3b, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/perraud/src/softwares/go/src/github.com/containous/yaegi/interp/interp.go:352 +0x237
github.com/containous/yaegi/interp.(*Interpreter).EvalWithContext.func1(0xc000044600, 0xc00014c1a0, 0xc0003420c0, 0x3b, 0xc00000e5c0, 0xc000076490)
	/home/perraud/src/softwares/go/src/github.com/containous/yaegi/interp/interp.go:435 +0x6e
created by github.com/containous/yaegi/interp.(*Interpreter).EvalWithContext
	/home/perraud/src/softwares/go/src/github.com/containous/yaegi/interp/interp.go:433 +0x14d

3:16: undefined: mimetype
```

**Comment by jlevesy** [comment]
Here's a simpler example that reproduces this issue, it seems to be related to recursive struct  types.

```
(19 ms) case13 λ cat main.go 
package main

type a struct {
	foo []*a
}

func main() {
	_ = a{foo: []*a{{}}}
	println("Success")
}
```

```
go run main.go
Success
```

```
yaegi main.go
(115 ms) case13 λ yaegi main.go 
run.go:1941: main.go:8:6: file
main.go:8:18: panic
reflect.Set: value of type []*struct { Xfoo []*interface {} } is not assignable to type []*interface {}
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc000326000, 0x40)
	/usr/lib/go/src/runtime/debug/stack.go:24 +0x9d
github.com/containous/yaegi/interp.(*Interpreter).Eval.func1(0xc00022bdf0)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/interp.go:330 +0xbc
panic(0xc9b420, 0xc0000a4640)
	/usr/lib/go/src/runtime/panic.go:969 +0x166
github.com/containous/yaegi/interp.runCfg.func1(0xc00030c0a0, 0xc00031dc00)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/run.go:114 +0x20c
panic(0xc9b420, 0xc0000a4640)
	/usr/lib/go/src/runtime/panic.go:969 +0x166
reflect.Value.assignTo(0xc0000b0400, 0xc0000a2840, 0x197, 0xdce00f, 0xb, 0xc0000b0380, 0x0, 0x413de4, 0xc00022bb90, 0xdce00f)
	/usr/lib/go/src/reflect/value.go:2403 +0x426
reflect.Value.Set(0xc0000b0380, 0xc0000a27e0, 0x197, 0xc0000b0400, 0xc0000a2840, 0x197)
	/usr/lib/go/src/reflect/value.go:1532 +0xbd
github.com/containous/yaegi/interp.doCompositeSparse.func1(0xc00030c0a0, 0xc0000b0480)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/run.go:1944 +0x13d
github.com/containous/yaegi/interp.runCfg(0xc00031dc00, 0xc00030c0a0)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/run.go:120 +0x6d
github.com/containous/yaegi/interp.(*Interpreter).run(0xc0000c61a0, 0xc00031cc00, 0xc00030c000)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/run.go:98 +0x239
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc0000c61a0, 0xc000314000, 0x65, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/interp.go:406 +0x52b
main.main()
	/home/jlevesy/code/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:143 +0x553
```

Hope this helps :smile:

**Comment by mvertes** [comment]
We are still investigating issues with support of recursive structs involving arrays (#687).

**Comment by nrwiersma** [comment]
Fixed by #742
